src/shogun-core-main/src/main/java/de/terrestris/shoguncore/util/interceptor/secure/WfsResponseInterceptor.java

```diff
@@ -180,6 +185,31 @@ public Response interceptDescribeFeatureType(MutableHttpServletRequest request,
         LOG.debug("Intercepting WFS DescribeFeatureType response");
         ByteArrayOutputStream bout = new ByteArrayOutputStream();
         try {
+            if (response.getHeaders().get("content-type").get(0).toLowerCase(Locale.ROOT).contains("json")) {
+                ObjectMapper objectMapper = new ObjectMapper();
+                JsonNode jsonObject = objectMapper.readTree(response.getBody());
+                String ns = jsonObject.get("targetPrefix").asText();
+                ArrayNode featureTypes = (ArrayNode) jsonObject.get("featureTypes");
+                List<Layer> allowedLayers = layerService.findAll();
```

Comment:
Please keep in mind that this might become very time consuming.

Reply:
better ideas?

Reply:
A custom SQL might be used instead.

Reply:
Thats possible. I think as this type of request happens rarely, it would be more important to make changes on requests like getmap